url78117c6033_BrianMoths_LittleManManager_tgz-pJ8-littlemangame_LittleManGameJ8/src/littlemangame/notebookdeveloper/submissioncontrols/SubmissionControlGui.java

```diff
@@ -0,0 +1,264 @@
+        speedPanel.setVisible(false);
+        submissionPanel.setVisible(true);
+        testButton.addActionListener(new ActionListener() {
+
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                showSpeedPanel();
+                speedControllerGui.resume();
+            }
+
+        });
+        submitButton.addActionListener(new ActionListener() {
+
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                final String resultString = SubmissionControlGui.this.notebookDeveloper.submitNotebookSolutionAttempt();
+                printResultMessage(resultString);
+            }
+
+        });
+        speedControllerGui.setEndTestAction(new ActionListener() {
+
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                showSubmissionPanel();
+                SubmissionControlGui.this.notebookDeveloper.endTest();
+            }
+
+        });
+    }
+
+    public void setEditMemoryAction(ActionListener l) {
+        final ActionListener[] originalActionListeners = editMemoryButton.getActionListeners();
+        for (ActionListener actionListener : originalActionListeners) {
+            editMemoryButton.removeActionListener(actionListener);
+        }
+        editMemoryButton.addActionListener(l);
+        for (ActionListener actionListener : originalActionListeners) {
+            editMemoryButton.addActionListener(actionListener);
+        }
+    }
+
+    public void removeEditMemoryAction(ActionListener actionListener) {
+        editMemoryButton.removeActionListener(actionListener);
+    }
+
+    protected void printResultMessage(String message) {
+        consoleTextArea.append(message);
+        consoleTextArea.setCaretPosition(0);
+    }
+
+    public final void showSpeedPanel() {
+        hidePanels();
+        speedPanel.setVisible(true);
+    }
+
+    public void registerSpeedController(SpeedController speedController) {
+        speedControllerGui.registerSpeedController(speedController);
+    }
+
+    public void registerNotebookDeveloper(T notebookDeveloper) {
+        this.notebookDeveloper = notebookDeveloper;
+    }
+
+    public final void showSubmissionPanel() {
+        hidePanels();
+        submissionPanel.setVisible(true);
+    }
+
+    protected void hidePanels() {
+        speedPanel.setVisible(false);
+        submissionPanel.setVisible(false);
+    }
+
+//    public SpeedControllerGui getSpeedControllerGui() {
+//        return speedControllerGui;
+//    }
+    final protected JLayeredPane getBaseLayeredPane() {
+        return baseLayeredPane;
+    }
+
+    final protected void replaceDummyPanel(Component newComponent) {
+        final GroupLayout groupLayout = (GroupLayout) baseLayeredPane.getLayout();
+
+        try {
+            groupLayout.replace(dummyPanel, newComponent);
+        } catch (IllegalArgumentException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)